### DISCUSSION TITLE
ConnectionFactory.MaxMessageSize not used

### DISCUSSION THREAD

**Discussion by pabermod** [rabbitmq/rabbitmq-dotnet-client#1251] [General]
Hi,

I have set the value of ConnectionFactory.MaxMessageSize (either with a value larger or smaller than the default value of the server).

If I sent a bigger file than the limit, it is always comparing against the value defined in the rabbitmq server (or the default). The field is not being used.

If this is the expected behaviour, what is the field MaxMessageSize used for?

**Comment by michaelklishin** [comment]
Clients are not allowed to use a value higher than that used by the server.

This value is meant to be used for validation and it was introduced about three months ago https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1218

**Reply by pabermod** [reply]
I tried to use this field to avoid sending a messge if its size is bigger. So there is no way to know from the client the max mesage size of the server?

**Comment by michaelklishin** [comment]
@lukebakken am I wrong about anything?

**Comment by lukebakken** [comment]
I don't believe client libraries have any knowledge of the server's maximum message size, because the value is not sent during the `connection.tune` phase of the AMQP handshake.

I'll have to check the code, but `MaxMessageSize` should be the maximum size of an AMQP message frame to prevent situations where a very large incoming or outgoing frame (or network bug) would cause the client library to try to request a huge amount of memory.

**Reply by lukebakken** [reply]
https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/6.x/projects/RabbitMQ.Client/client/impl/Frame.cs#L247-L251
